```
======================================================================
ERROR: test suite for <class 'testapps.test_pillowtop.tests.test_case_pillow.CasePillowTest'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python3.6/site-packages/nose/suite.py", line 210, in run
    self.setUp()
  File "/vendor/lib/python3.6/site-packages/nose/suite.py", line 293, in setUp
    self.setupContext(ancestor)
  File "/vendor/lib/python3.6/site-packages/nose/suite.py", line 316, in setupContext
    try_run(context, names)
  File "/vendor/lib/python3.6/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/mnt/commcare-hq-ro/testapps/test_pillowtop/tests/test_case_pillow.py", line 32, in setUpClass
    delete_es_index(CASE_SEARCH_INDEX_INFO.index)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 33, in inner
    return fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/elastic.py", line 27, in delete_es_index
    es.indices.delete(index=es_index)
  File "/vendor/lib/python3.6/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/vendor/lib/python3.6/site-packages/elasticsearch/client/indices.py", line 198, in delete
    params=params)
  File "/vendor/lib/python3.6/site-packages/elasticsearch/transport.py", line 307, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/vendor/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py", line 93, in perform_request
    self._raise_error(response.status, raw_data)
  File "/vendor/lib/python3.6/site-packages/elasticsearch/connection/base.py", line 105, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
elasticsearch.exceptions.NotFoundError: TransportError(404, 'IndexMissingException[[test_case_search_2018-05-29] missing]')
```

I think you want `ensure_index_deleted`

--- a/testapps/test_pillowtop/tests/test_case_pillow.py
+++ b/testapps/test_pillowtop/tests/test_case_pillow.py
@@ -17,7 +17,7 @@ from corehq.form_processor.tests.utils import (
 )
 from corehq.pillows.mappings.case_mapping import CASE_INDEX_INFO
 from corehq.pillows.mappings.case_search_mapping import CASE_SEARCH_INDEX_INFO
-from corehq.util.elastic import delete_es_index, ensure_index_deleted
+from corehq.util.elastic import ensure_index_deleted
 from corehq.util.test_utils import create_and_save_a_case, trap_extra_setup
 from testapps.test_pillowtop.utils import process_pillow_changes
 
@@ -28,8 +28,8 @@ class CasePillowTest(TestCase):
     @classmethod
     def setUpClass(cls):
         super(CasePillowTest, cls).setUpClass()
-        delete_es_index(CASE_INDEX_INFO.index)
-        delete_es_index(CASE_SEARCH_INDEX_INFO.index)
+        ensure_index_deleted(CASE_INDEX_INFO.index)
+        ensure_index_deleted(CASE_SEARCH_INDEX_INFO.index)
 
     def setUp(self):
         super(CasePillowTest, self).setUp()

